internal/pedersen/commitment.go

```diff
@@ -0,0 +1,151 @@
+package pedersen
+
+import (
+	"crypto/sha256"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/lightningnetwork/lnd/fn"
+	"github.com/lightningnetwork/lnd/input"
+)
+
+var (
+	// blankMessage is the "empty" value for a commitment message.
+	blankMessage [sha256.Size]byte
+
+	// DefaultNUMs is the default NUMs key used for Pedersen commitments.
+	DefaultNUMs = input.TaprootNUMSKey
+
+	// one is the value 1 as a scalar value.
+	one = new(btcec.ModNScalar).SetInt(1)
+)
+
+// Opening is the opening to a Pedersen commitment. It contains a message, and
+// an optional mask. If the mask is left off, then the commitment will lose its
+// hiding property (two identical messages will map to the same point), but the
+// binding property is kept.
+type Opening struct {
+	// Msg is the message that was committed to.
+	Msg [sha256.Size]byte
+
+	// Mask is the mask used to blind the message. This is typically
+	// referred to as `r` in the Pedersen commitment literature.
+	//
+	// We make this optional, as without it we'll default to no value, which
+	// means that the commitment loses the hiding attribute, but still
+	// remains computationally binding.
+	Mask fn.Option[[sha256.Size]byte]
+
+	// NUMs is an optional value that should be used to verify the
+	// commitment if a custom NUMs point was used.
+	NUMs fn.Option[btcec.PublicKey]
+}
+
+// Commitment is a Pederson commitment of the form: m*G + r*H, where:
```

Comment:
```suggestion
// Commitment is a Pedersen commitment of the form: m*G + r*H, where:
```
branch is also labeled `son` rather than `sen`